web/src/components/PackagesList.vue

```diff
@@ -39,130 +34,80 @@ export default {
       let height = this.$refs.innerContainer.getBoundingClientRect().height;
+
       if (bottom > height - 200 && this.morePackagesStatus == "loaded") {
         this.page += 1;
-        this.appendPackages();
+        this.queryNextPackagesPage();
       }
     },
-    createQuery() {
-      if (this.queryUrl == null) {
-        return new Promise((resolve, reject) => {
-          axios
-            .get(this.source.url)
-            .then(response => {
-              let resource = response.data.resources.find(x =>
-                x["@type"].includes("SearchQueryService")
-              );
-              if (resource != null) this.queryUrl = resource["@id"];
-              if (this.queryUrl == null) reject();
-              this.createQuery()
-                .then(response => resolve(response))
-                .catch(error => reject(error));
-            })
-            .catch(error => {
-              reject(error);
-            });
-        });
-      }
+    refresh() {
+      if (!this.source) return;
+
+      this.status = "loading";
 
-      return axios.get(this.queryUrl, {
-        params: {
-          q: this.filter,
-          take: this.pageSize,
-          skip: this.page * this.pageSize
-        }
+      this.page = 0;
+      this.selectPackage(null);
+      this.packages = null;
+
+      this.$emit("refreshPackages", {
+        source: this.source,
+        page: this.page,
+        pageSize: this.pageSize,
+        filter: this.filter
       });
     },
-    appendPackages() {
-      if (this.source == null) return;
+    listPackages(packages) {
+      this.packages = packages;
+      this.status = "loaded";
+    },
+    queryNextPackagesPage() {
+      if (!this.source) return;
+
       this.morePackagesStatus = "loading";
-      this.createQuery()
-        .then(response => {
-          this.morePackagesStatus = "loaded";
-          if (response.data && response.data.data.length > 0)
-            response.data.data.forEach(x => this.packages.push(x));
-          else this.morePackagesStatus = "all";
-        })
-        .catch(err => {
-          this.morePackagesStatus = "error";
-        });
+
+      this.$emit("queryPackagesPage", {
+        source: this.source,
+        page: this.page,
+        pageSize: this.pageSize,
+        filter: this.filter
+      });
     },
-    refresh() {
-      if (this.source == null) return;
-      this.page = 0;
-      this.selectPackage(null);
-      this.packages = null;
-      this.status = "loading";
-      this.createQuery()
-        .then(response => {
-          if (response.data) this.packages = response.data.data;
-          this.status = "loaded";
-        })
-        .catch(err => {
-          this.status = "error";
-        });
+    appendPackages(packages) {
+      if (packages.length > 0) {
+        this.morePackagesStatus = "loaded";
+        this.packages = packages.forEach(p => this.packages.push(p));
```

Comment:
should be `packages.forEach(p => this.packages.push(p));` instead of `this.packages = packages.forEach(p => this.packages.push(p));`